### PR TITLE
Fix passing --interactive flag to nuget

### LINKED ISSUE #6020: dotnet new install ignores --interactive option and therefore cannot authenticate with Azure Artifacts Credential Provider
### Describe the bug
When attempting to install a template from a private Azure Artifacts feed that has not yet been authenticated via the Azure Artifacts Credential Provider, an error is returned and the package fails to install.

### To Reproduce
Prerequisites
- .NET 7 SDK (7.0.102 in my case)
- Azure Artifacts Credential Provider (1.0.0)

1. Add a new private registry to your sources list:

`
dotnet nuget add source https://myteam.pkgs.visualstudio.com/_packaging/myTeam/nuget/v3/index.json -n myTeam
`

2. Attempt to install a template from that new source (that you have not yet acquired credentials for)

`
dotnet new install My.Template --interactive
`

**Expected:** a prompt to authenticate via microsoft devicelogin
**Actual:** a warning that the credential provider could not acquire credentials and an error stating the package does not exist.

### Exceptions (if any)
Relevant logs from using the --diagnostics option:

```
[2023-01-31 15:16:25.652] [Debug] [NuGetLogger] => [Execute]: Searching for My.Template in https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json.
[NuGet Manager] [DEBUG] Using C:\Users\Me\.nuget\plugins\netcore\CredentialProvider.Microsoft\CredentialProvider.Microsoft.dll as a credential provider plugin.
[NuGet Manager] [Verbose]     [CredentialProvider.151626]Running in plug-in mode
[NuGet Manager] [Verbose]     [CredentialProvider.151626]Command-line v1.0.0+94af8126cf520805dd7e213b92cdecd9d8728d1f: C:\Users\Me\.nuget\plugins\netcore\CredentialProvider.Microsoft\CredentialProvider.Microsoft.dll -Plugin
[NuGet Manager] [Verbose]     [CredentialProvider]Handling 'Request' 'GetAuthenticationCredentials'. Time elapsed in ms: 2 - Payload: {"Uri":"https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json","IsRetry":false,"IsNonInteractive":true,"CanShowDialog":false}
[NuGet Manager] [Verbose]     [CredentialProvider]Handling auth request, Uri: https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json, IsRetry: False, IsNonInteractive: True, CanShowDialog: False
[NuGet Manager] [Verbose]     [CredentialProvider]URI: https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json
[NuGet Manager] [Verbose]     [CredentialProvider]VstsBuildTaskServiceEndpointCredentialProvider - This credential provider must be run under the Team Build tasks for NuGet with external endpoint credentials. Appropriate environment variable needs to be set.
[NuGet Manager] [Verbose]     [CredentialProvider]Skipping NuGetCredentialProvider.CredentialProviders.VstsBuildTaskServiceEndpoint.VstsBuildTaskServiceEndpointCredentialProvider, cannot provide credentials for https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json
[NuGet Manager] [Verbose]     [CredentialProvider]VstsBuildTaskCredentialProvider - This credential provider must be run under the Team Build tasks for NuGet. Appropriate environment variables must be set.
[NuGet Manager] [Verbose]     [CredentialProvider]Skipping NuGetCredentialProvider.CredentialProviders.VstsBuildTask.VstsBuildTaskCredentialProvider, cannot provide credentials for https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json
[NuGet Manager] [Verbose]     [CredentialProvider]GET https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json
[NuGet Manager] [Verbose]     [CredentialProvider]VstsCredentialProvider - Detected a hosted Azure DevOps Service.
[NuGet Manager] [Verbose]     [CredentialProvider]Using NuGetCredentialProvider.CredentialProviders.Vsts.VstsCredentialProvider to try to get credentials for https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json.
[NuGet Manager] [Verbose]     [CredentialProvider]IsRetry: False
[NuGet Manager] [Verbose]     [CredentialProvider]Could not find cached SessionToken for https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json
[NuGet Manager] [Verbose]     [CredentialProvider]GET https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json
[NuGet Manager] [Verbose]     [CredentialProvider]Found AAD Authority from 401 headers: https://login.windows.net/guid
[NuGet Manager] [Verbose]     [CredentialProvider]VstsCredentialProvider - Using AAD authority: https://login.windows.net/guid
[NuGet Manager] [Verbose]     [CredentialProvider]VstsCredentialProvider - Attempting to acquire bearer token using provider 'ADAL Cache'
[NuGet Manager] [Verbose]     [CredentialProvider]VstsCredentialProvider - Bearer token provider 'ADAL Cache' didn't acquire a token
[NuGet Manager] [Verbose]     [CredentialProvider]VstsCredentialProvider - Attempting to acquire bearer token using provider 'ADAL Windows Integrated Authentication'
[NuGet Manager] [Warning] The plugin credential provider could not acquire credentials. Authentication may require manual action. Consider re-running the command with --interactive for `dotnet`, /p:NuGetInteractive="true" for MSBuild or removing the -NonInteractive switch for `NuGet`
[NuGet Manager] [Verbose]     [CredentialProvider]VstsCredentialProvider - Bearer token provider 'ADAL Windows Integrated Authentication' failed with exception:\nMicrosoft.IdentityModel.Clients.ActiveDirectory.AdalException: Integrated Windows Auth is not supported for managed users. See https://aka.ms/adal-iwa for details.
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenIWAHandler.PreTokenRequestAsync()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows.AcquireTokenHandlerBase.RunAsync()
   at Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext.AcquireTokenCommonAsync(String resource, String clientId, IntegratedWindowsAuthInput iwaInput)
   at Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContextIntegratedAuthExtensions.AcquireTokenAsync(AuthenticationContext ctx, String resource, String clientId, UserCredential userCredential)
   at NuGetCredentialProvider.CredentialProviders.Vsts.AdalTokenProvider.AcquireTokenWithWindowsIntegratedAuth(CancellationToken cancellationToken) in D:\a\_work\1\s\CredentialProvider.Microsoft\CredentialProviders\Vsts\AdalTokenProvider.cs:line 122
   at NuGetCredentialProvider.CredentialProviders.Vsts.WindowsIntegratedAuthBearerTokenProvider.GetTokenAsync(Uri uri, CancellationToken cancellationToken) in D:\a\_work\1\s\CredentialProvider.Microsoft\CredentialProviders\Vsts\BearerTokenProviders.cs:line 58
   at NuGetCredentialProvider.CredentialProviders.Vsts.VstsCredentialProvider.HandleRequestAsync(GetAuthenticationCredentialsRequest request, CancellationToken cancellationToken) in D:\a\_work\1\s\CredentialProvider.Microsoft\CredentialProviders\Vsts\VstsCredentialProvider.cs:line 117
        ErrorCode: integrated_windows_auth_not_supported_managed_user
[NuGet Manager] [Verbose]     [CredentialProvider]VstsCredentialProvider - Not running bearer token provider 'ADAL UI'
[NuGet Manager] [Verbose]     [CredentialProvider]VstsCredentialProvider - Not running bearer token provider 'ADAL Device Code'
[NuGet Manager] [Verbose]     [CredentialProvider]VstsCredentialProvider - Could not obtain credentials for https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json
[NuGet Manager] [Verbose]     [CredentialProvider]Unable to acquire credentials.
[NuGet Manager] [Verbose]     [CredentialProvider]Sending response: 'Request' 'GetAuthenticationCredentials'. Time elapsed in ms: 416
[NuGet Manager] [Verbose]     [CredentialProvider]Time elapsed in milliseconds after sending response 'Request' 'GetAuthenticationCredentials': 418
[2023-01-31 15:16:32.353] [Debug] [NuGetLogger] => [Execute]: Failed to read package information from NuGet source https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json.
[2023-01-31 15:16:32.361] [Debug] [NuGetLogger] => [Execute]: Details: NuGet.Protocol.Core.Types.FatalProtocolException: Unable to load the service index for source https://pkgs.dev.azure.com/myteam/_packaging/myTeam/nuget/v3/index.json.
 ---> System.Net.Http.HttpRequestException: Response status code does not indicate success: 401 (Unauthorized).
My.Template could not be installed, no NuGet feeds are configured or they are invalid.
[2023-01-31 15:16:32.410] [Debug] [Template Engine] => [Execute]: Execute finished, took 6858 ms

For details on the exit code, refer to https://aka.ms/templating-exit-codes#106

```

Note the options "IsNonInteractive":true and "CanShowDialog":false are being passed to the credential provider plugin which doesn't appear to be correct given I've used the --interactive option.

### PR DESCRIPTION
**Pull request by YuliiaKovalova**
## Problem
Fixes https://github.com/dotnet/templating/issues/6020

## Solution
After major refactoring between net 6 and net 7 the logic of NuGetPackageDownloader instantiation,  passing `--interactive` flag was missed for the install command.

### CODE REVIEW

File: src/Cli/dotnet/commands/dotnet-workload/WorkloadInfoHelper.cs [hunk 1]
```diff
@@ -21,6 +23,7 @@ internal class WorkloadInfoHelper : IWorkloadInfoHelper
         private readonly string _targetSdkVersion;
 
         public WorkloadInfoHelper(
+            ParseResult parseResult,
```
Comment (vlada-shubina) [inline src/Cli/dotnet/commands/dotnet-workload/WorkloadInfoHelper.cs:26]: it might be ok just to pass `bool interactive` instead of whole parse result.

**Review by vlada-shubina — COMMENTED** [review]
This looks good to me, @dsplaisted @marcpopMSFT could you please review workload part too?

File: src/Cli/dotnet/commands/dotnet-workload/WorkloadInfoHelper.cs (hunk 1 above)
Comment (YuliiaKovalova) [reply src/Cli/dotnet/commands/dotnet-workload/WorkloadInfoHelper.cs]: fixed

**Review by dsplaisted — APPROVED** [review]
Workload changes look good.

**Pull request merged**